### PR TITLE
feat: add tracef

### PR DESCRIPTION
**Pull request by philhuan**
add tracef

### CODE REVIEW

File: pkg/entity/run/run_test.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+)
+
+func TestDefExecuteContext_Tracef(t *testing.T) {
+	tests := []struct {
+		name      string
+		operation func(e *DefExecuteContext)
+		wantLogs  []interface{}
+	}{
+		{
+			name: "",
```
Comment (ShiningRush) [inline pkg/entity/run/run_test.go:17]: maybe we should give a name?

File: pkg/entity/run/run.go [hunk 2]
```diff
@@ -71,6 +74,32 @@ func (e *DefExecuteContext) Trace(msg string, opt ...TraceOp) {
 	e.trace(msg, opt...)
 }
 
+// Tracef
+func (e *DefExecuteContext) Tracef(msg string, a ...interface{}) {
+	args, ops := splitArgsAndOpt(a...)
+	e.trace(fmt.Sprintf(msg, args...), ops...)
+}
+
+func splitArgsAndOpt(a ...interface{}) ([]interface{}, []TraceOp) {
```
Comment (ShiningRush) [inline pkg/entity/run/run.go:83]: This implement limit user must write trace's options at end,  we should highlight it in method's description.
Or we also could implement `Tracef` as beblow:
```golang

func splitArgsAndOpt(a ...interface{}) (args []interface{},opts []TraceOp)
for _, item := range a {
    if _, ok := item.(TraceOp); ok {
      opts = append(opts, item)
      continue
   }
   args = append(args, item)
}
return
}
```

so developer can ignore the order of parameter. how do you think?

**Review by ShiningRush — CHANGES_REQUESTED** [review]

File: pkg/entity/run/run.go (hunk 2 above)
Comment (philhuan) [reply pkg/entity/run/run.go:83]: It is customary to put opt ​​at the end. In addition, it can have better performance when placed at the end.

**Review by ShiningRush — APPROVED** [review]
cool~

**Pull request merged**